Clarify that non-JARM error response may be sent even when JARM requested (#360)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

The Response Mode `direct_post.jwt` causes the Wallet to send the Authorization Response using an HTTP POST request instead of redirecting back to the Verifier as defined in (#response_mode_post). The Wallet adds the `response` parameter containing the JWT as defined in Section 4.1. of [@!JARM] and (#jarm) in the body of an HTTP POST request using the `application/x-www-form-urlencoded` content type. The names and values in the body MUST be encoded using UTF-8.

> If a Wallet is unable to send a JARM response it MAY send an error response as per (#response_mode_post).

Reviewer: [Response Mode "direct_post"](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-wg-draft.html#name-response-mode-direct_post) does not really define error responses clearly....

I might have misunderstood the discussion in the issue, but I think this sentence should be clearer?

```suggestion
If a Wallet is unable to generate a JARM response, it MAY send an error response without using JARM.
```

Author: > [Response Mode "direct_post"](https://openid.github.io/OpenID4VP/openid-4-verifiable-presentations-wg-draft.html#name-response-mode-direct_post) does not really define error responses clearly....

It is where the non-JARM way of sending responses is defined, and It has an example of an error being sent at least. I'd prefer to keep the reference to make it clear what is meant. Happy to change send to generate though, does this work for you:

```suggestion
If a Wallet is unable to generate a JARM response, it MAY send an error response as per (#response_mode_post).
```

Reviewer: umm, i really think it should be explicitly stated that an error response can be without using JARM, which I think is the point of this PR, right>

Author: Yes - that is the point, the `(#response_mode_post)` is a link to the non-JARM response definition.

Reviewer: yes, so if that's the point of that PR, I would like to see the language to be more explicit

Author: ```suggestion
If a Wallet is unable to generate a JARM response, it MAY send a non-JARM error response as per (#response_mode_post).
```

@Sakurann does explicitly mentioning non-JARM like this this solve it?

Reviewer: what is a non-JARM error response? why can't we say "error response without using JARM."

Author: ```suggestion
If a Wallet is unable to generate a JARM response, it MAY send an error response without using JARM as per (#response_mode_post).
```

Changed to use Kristina's suggested wording.
